File: src/waypalm-site/src/main/java/waypalm/site/web/extensions/VisitorExtender.java
package waypalm.site.web.extensions;

import org.springframework.ui.ModelMap;
import org.springframework.web.context.request.WebRequest;
import waypalm.common.web.extensions.ModelExtension;
import waypalm.domain.entity.Profile;
import waypalm.site.model.view.UserName;
import waypalm.site.services.ModelViewer;
import waypalm.site.services.UserAccess;
import waypalm.site.services.UserService;

import javax.inject.Inject;

@ModelExtension("eVisitor")
public class VisitorExtender extends Extender<VisitorExtender.Content> {
    @Inject
    ModelViewer modelViewer;
    @Inject
    UserService userService;
    @Inject
    UserAccess auth;

    @Override
    protected VisitorExtender.Content populate(WebRequest request, VisitorExtender.Content extension, ModelMap model) {
        Profile user = auth.loadCurrentProfile();
        extension = instantiateIfNull(extension, VisitorExtender.Content.class);
        if (user != null) {
            extension.user = modelViewer.createUserName(user);
        }
        return extension;
    }

    public static class Content {
        private UserName user;

        public UserName getUser() {
            return user;
        }

        public boolean isLogged() {
            return user != null;
        }
    }
}

File: src/waypalm-site/src/main/java/waypalm/site/web/extensions/VisitorExtender.kt
package waypalm.site.web.extensions

import waypalm.common.web.extensions.ModelExtension
import javax.inject.Inject
import waypalm.site.services.ModelViewer
import waypalm.site.services.UserAccess
import waypalm.site.services.UserService
import org.springframework.web.context.request.WebRequest
import org.springframework.ui.ModelMap
import waypalm.site.model.view.UserName

ModelExtension("eVisitor")
public class VisitorExtender [Inject](
        val modelViewer: ModelViewer,
        val userService: UserService,
        val auth: UserAccess
)
: Extender<VisitorExtender.Content>()
{
    public  class Content {
        var user: UserName? = null

        fun isLogged(): Boolean {
            return user != null
        }
    }

    protected override fun populate(request: WebRequest, extension: VisitorExtender.Content?, model: ModelMap): VisitorExtender.Content {
        val user = auth.loadCurrentProfile()
        var result = instantiateIfNull(extension, javaClass<VisitorExtender.Content>())
        if (user != null) {
            result.user = modelViewer.createUserName(user)
        }
        return result ;
    }
}